### PR TITLE
Check for sample IDs characters in samplesheet and report line numbers when erroring

### LINKED ISSUE #338: the sample field from the samplesheet is too lenient
### Description of the bug

@lindenb reported a bug with `/LAB-DATA/BiRD/shares/ITX/ITX_GLICID/commun/PF/NTS-333_NGS_Rimbert_ATAC-Seq/fastq/P1_S1_B1_S1` as a sample id, that created files in this position, because this id was used as prefix...

I do believe that we should not permit such a sample id to exist.

cf https://nfcore.slack.com/archives/CE5EL6326/p1696864888017299

### Command used and terminal output

_No response_

### Relevant files

_No response_

### System information

_No response_

### LINKED ISSUE #370: Add line number when reporting errors in check_samplesheet.py
### Description of feature

See https://github.com/nf-core/chipseq/pull/392

### PR DESCRIPTION
**Pull request by JoseEspinosa**
Fixes #338 
Closes #370 

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [x] Make sure your code lints (`nf-core lint`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [x] `CHANGELOG.md` is updated.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -12,6 +12,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Updated pipeline template to [nf-core/tools 2.14.1](https://github.com/nf-core/tools/releases/tag/2.14.1)
 - [[#359](https://github.com/nf-core/atacseq/issues/359)] - Fix `--save_unaligned` description in schema.
 - [[#344](https://github.com/nf-core/atacseq/issues/344)] - Fix memory issues when sorting merged replicates after `bedtools genomecov`.
+- [[#338](https://github.com/nf-core/atacseq/issues/338)] - Check that samplesheet samples IDs do only have alphanumeric characters, dots, dashes or underscores.
+- [[#370](https://github.com/nf-core/atacseq/issues/370)] - Adding line numbers to errors messages in `bin/check_samplesheet.py`
```
Comment (maxulysse) [inline CHANGELOG.md:16]: so you link issues and not PRs?

**Review by maxulysse — APPROVED** [review]
LGTM.
What's the plan towards switching to nf-validation/nf-schema?

File: CHANGELOG.md (hunk 1 above)
Comment (JoseEspinosa) [reply CHANGELOG.md:16]: Both, if you look to the rnaseq changelog you also do 😉

**Comment by JoseEspinosa** [comment]
> LGTM. What's the plan towards switching to nf-validation/nf-schema?

We are doing some validations inside the samplesheet apart from the normal ones. But I was discussing with @mirpedrol and @bjlang and we might give it a try for the release we are planning. We probably will start with nf-core/chipseq since there the controls are mandatory and makes it easier to implement

**Pull request merged**